pkg/sql/opt/exec/execbuilder/tests/5node/5node_test: TestExecBuild_distsql_scan failed

pkg/sql/opt/exec/execbuilder/tests/5node/5node_test.TestExecBuild_distsql_scan [failed](https://mesolite.cluster.engflow.com/invocations/default/701a9a01-6492-4d74-9bc0-d310d311da25?testReportRun=4&testReportShard=15&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4ZWNidWlsZGVyL3Rlc3RzLzVub2RlOjVub2RlX3Rlc3Q=) on release-25.2 @ [91889f0cfb89099f32c4f9ea7dd1108c00d0e395](https://github.com/cockroachdb/cockroach/commits/91889f0cfb89099f32c4f9ea7dd1108c00d0e395):


```
              KV pairs read: 20
              KV bytes read: 80 B
              KV gRPC calls: 10
              estimated max memory allocated: 0 B
              estimated row count: 10 (10% of the table; stats collected <hidden> ago)
              table: abc@abc_pkey
              spans: [/0 - /99]
              limit: 10
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyMktFq20wQhe__pxjmKgEFr_7elIVCG9ulpnYcZJNSWhNGq4mzWNKquyNsY_xYfYE-WVnJTtrSQPdCaM5-e-asRgcM30rUuBhPx8MlELzP5jOg3MCnD-NsDBcEX1ulXvEbUJfw7mb0pBhIlbqEeTYaZ3D9GQimk9lkCanCBGtX8A1VHFB_wRRXCTbeGQ7B-SgdOmBS7FCrBG3dtBLlVYLGeUZ9QLFSMmpcUl5yxlSwH0TjgoVs2dlSbt5Sbu6bDe8xwaEr26oOGggTXDQUXwfqapCqeG5qKyt9to93ELsHDXXal57X1kVeOEgvia1Yg_rxPZwQtw1QsHEFF_rJJ98LB_BMhYbXCq57dZ3dDsFQWYZnsiHrz-T_UZvdDYcQhBswrq0FLngnA1vLpQY1eAaYNy8BFe2g4sr5PVBZOkMSo_UpchLzyAFcK00rGiLfXeEspApXxwT78vTtg9CaUae_DGsyQq2Oyb_PK-PQuDrwb6N6qZP6o9NVelwlyMWa-58kuNYbvvXOdGxfzjujTig4SL-b9sWkPm8F8UxVH3-V4EPptve2QI3qtK7-8jgvjAdoHeLFFo9u29ku902M9UBl4ARntOERC_vK1jaINajFt3w8_vczAAD__xOuBa4=
            
        but found (query options: "") :
            planning time: 10µs
            execution time: 100µs
            distribution: <hidden>
            vectorized: <hidden>
            plan type: custom
            rows decoded from KV: 10 (80 B, 20 KVs, 10 gRPC calls)
            maximum memory usage: <hidden>
            DistSQL network usage: <hidden>
            regions: <hidden>
            isolation level: serializable
            priority: normal
            quality of service: regular
            ·
            • scan
              sql nodes: <hidden>
              kv nodes: <hidden>
              regions: <hidden>
              actual row count: 10
              KV time: 0µs
              KV rows decoded: 10
              KV pairs read: 20
              KV bytes read: 80 B
              KV gRPC calls: 10
              estimated max memory allocated: 0 B
              missing stats
              table: abc@abc_pkey
              spans: [/0 - /99]
              limit: 10
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyMktFq20wQhe__pxjmKgEFr_7elIVCG9ulpnYcZJNSWhNGq4mzWNKquyNsY_xYfYE-WVnJTtrSQPdCaM5-e-asRgcM30rUuBhPx8MlELzP5jOg3MCnD-NsDBcEX1ulXvEbUJfw7mb0pBhIlbqEeTYaZ3D9GQimk9lkCanCBGtX8A1VHFB_wRRXCTbeGQ7B-SgdOmBS7FCrBG3dtBLlVYLGeUZ9QLFSMmpcUl5yxlSwH0TjgoVs2dlSbt5Sbu6bDe8xwaEr26oOGggTXDQUXwfqapCqeG5qKyt9to93ELsHDXXal57X1kVeOEgvia1Yg_rxPZwQtw1QsHEFF_rJJ98LB_BMhYbXCq57dZ3dDsFQWYZnsiHrz-T_UZvdDYcQhBswrq0FLngnA1vLpQY1eAaYNy8BFe2g4sr5PVBZOkMSo_UpchLzyAFcK00rGiLfXeEspApXxwT78vTtg9CaUae_DGsyQq2Oyb_PK-PQuDrwb6N6qZP6o9NVelwlyMWa-58kuNYbvvXOdGxfzjujTig4SL-b9sWkPm8F8UxVH3-V4EPptve2QI3qtK7-8jgvjAdoHeLFFo9u29ku902M9UBl4ARntOERC_vK1jaINajFt3w8_vczAAD__xOuBa4=
            
[06:40:53] --- done: /mnt/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_scan with config 5node: 7 tests, 1 failures
    logic.go:4326: 
        /mnt/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_scan:68: error while processing
    logic.go:4326: /mnt/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/5node/5node_test_/5node_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_scan:68: too many errors encountered, skipping the rest of the input
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestExecBuild_distsql_scan656314726
--- FAIL: TestExecBuild_distsql_scan (1.98s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=15</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_scan.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58496

--- a/pkg/sql/opt/exec/execbuilder/testdata/distsql_scan
+++ b/pkg/sql/opt/exec/execbuilder/testdata/distsql_scan
@@ -31,6 +31,10 @@ start_key           end_key       replicas  lease_holder
 statement ok
 ANALYZE abc
 
+# Clear the stat cache so that the new statistic is guaranteed to be used.
+statement ok
+SELECT crdb_internal.clear_table_stats_cache();
+
 query T
 EXPLAIN ANALYZE (DISTSQL) SELECT a FROM abc WHERE a >= 0 AND a < 100 ORDER BY a LIMIT 10
 ----

